.env

```diff
@@ -50,7 +50,7 @@ SPDK_URL="https://spdk.io"
 
 SPDK_PKGDEP_ARGS="--rbd"
 # check spdk/configure --help
-SPDK_CONFIGURE_ARGS="--with-rbd --disable-tests --disable-unit-tests --disable-examples --enable-debug"
+SPDK_CONFIGURE_ARGS="--with-rbd --with-idxd --disable-tests --disable-unit-tests --disable-examples --enable-debug"
```

Comment:
This won't work on older CPU, please detect it and only is supported enable it

Reply:
As we build right now, it might not work on any CPU. To eliminate crashes on some machine we build the SPDK using x86-64-v2 architecture. This instructs the compiler not to use instructions which are not available on v2. We need to verify that the instructions used here are supported on v2 CPUs.

Reply:
@CongMinYin  @gbregman the question is what happens if one one hand we build to x86-64-v2, but on the other hand we specify this flag (--with-idxd). I assume that the SPDK build will ignore the --with-idxd because it is not supported on x86-64-v2?

Reply:
@caroav this can easily verified. We can just disassemble the SPDK binary and see if this instruction was used or not. This is what I did when we had the issue with RDSEED.

Reply:
@caroav I think we shuold check `enable_dsa` in `ceph-nvmeof.conf`. But I don't how to write something like "if else" in `.env` or `Dockerfile`. It seems like you're saying the x86-64-v2 instruction set. But in reality, DSA is a DMA engine, an accelerator only integrated into the new 4th Gen CPU now. 

`idxd`  is a kernel driver.  I have tried to configure `idxd` on server with AMD CPU, it will report an error and exit. So it's best to check the `enable_dsa` option in the `ceph-nvmeof.conf`. Provide different parameters in .`env` or `Dockerfile`. But I couldn't find a similar code block. Do you know what to do?
Run configure in a server with AMD CPU:
```
root@amd-server4:~/spdk# ./configure  --with-idxd
ERROR: IDXD cannot be used due to CPU incompatibility.
```
SPDK configure code to detect CPU:
https://github.com/spdk/spdk/blob/dfb2950fd5b13ba0a2a2b533e3e534202a89de74/configure#L741

Reply:
I think the IDXD code itself is safe for all hardware platforms:

1. It is scanning for a specific PCIe device (PCI_DEVICE_ID_INTEL_IDXD) and will not initialize if this device is not found. Older Intel CPUs and AMD CPUs will not have this device.
2. It needs to use the movdir64b instruction to pass work to the hardware, this is only supported on new Intel CPUs. However this instruction is only ever executed if the PCIe device is found. Due to lack of compiler support this instruction is being generated using a sequence of bytes in assembly code:

<pre>
static inline void movdir64b(void *dst, const void *src)
{
        asm volatile(".byte 0x66, 0x0f, 0x38, 0xf8, 0x02"
                     : "=m"(*(char *)dst)
                     : "d"(src), "a"(dst));
}
</pre>

So building the code with --with-idxd will include a new assembly instruction, but it will only be executed on CPUs that support it because of the PCIe device check. Its very hard to prove this other than by code review.

The configure script for SPDK is only permitting the code to be compiled with --with-idxd if it is compiled on an Intel CPU. This means you can't use an AMD server as the build machine. The script is not checking the CPU version - so you can build the code with support for IDXD on an old Intel CPU.

I suggest that the check in the configure script is overly restrictive, and perhaps should be reduce to a warning rather than an error as the code may be built on a system with different capabilities to the system it is run on. It might also be better to check for the presence of the PCIe device rather than the CPU vendor to determine whether the build server supports idxd.

Reply:
Good catch, we need support for the code to be compiled for non-Intel CPU

Reply:
@caroav Is our solution to try submitting a PR to modify the SPDK's config script, or check the `enable_dsa` parameter in this `ceph-nvmeof` project?

Reply:
Hi @caroav @oritwas,  I have submit a pr to solve the platforms errors: https://review.spdk.io/gerrit/c/spdk/spdk/+/24172. It merged. SPDK master branch and later versions won't met this platform errors.